iPetCare/Application/Services/ExaminationTypesService.cs

```diff
@@ -0,0 +1,167 @@
+            }
+
+            if (currentUserName == null)
+                return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var examinationType = new ExaminationType()
+            {
+                Name = dto.Name,
+                SpeciesId = dto.SpeciesId
+            };
+
+            Context.ExaminationTypes.Add(examinationType);
+            int result = await Context.SaveChangesAsync();
+
+            if (result > 0)
+            {
+                var responseDto = new ExaminationTypesCreateDtoResponse()
+                {
+                    Name = examinationType.Name,
+                    SpeciesId = examinationType.SpeciesId,
+                    Id = examinationType.Id
+                };
+
+                return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.OK, responseDto);
+            }
+
+            return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.BadRequest);
+        }
+
+        public async Task<ServiceResponse<ExaminationTypesGetAllDtoResponse>> GetAllExaminationTypesAsync()
+        {
+            var currentUserName = CurrentlyLoggedUserName;
+
+            if (currentUserName == null)
+                return new ServiceResponse<ExaminationTypesGetAllDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var currentUser = await UserManager.FindByNameAsync(currentUserName);
+            if (currentUser == null)
+                return new ServiceResponse<ExaminationTypesGetAllDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var examinationType = await Context.ExaminationTypes.ToListAsync();
+
+            var dto = new ExaminationTypesGetAllDtoResponse();
+            dto.ExaminationTypes = Mapper.Map<List<ExaminationTypesDetailGetAllDtoResponse>>(examinationType);
+
+            return new ServiceResponse<ExaminationTypesGetAllDtoResponse>(HttpStatusCode.OK, dto);
+        }
+
+        public async Task<ServiceResponse<ExaminationTypeGetDtoResponse>> GetExaminationTypeAsync(int examinationTypeId)
+        {
+            var currentUserName = CurrentlyLoggedUserName;
+
+            if (currentUserName == null)
+                return new ServiceResponse<ExaminationTypeGetDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var currentUser = await UserManager.FindByNameAsync(currentUserName);
+            if (currentUser == null)
+                return new ServiceResponse<ExaminationTypeGetDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var examinationType = await Context.ExaminationTypes.FindAsync(examinationTypeId);
+            if (examinationType == null)
+                return new ServiceResponse<ExaminationTypeGetDtoResponse>(HttpStatusCode.BadRequest, "Nie istnieje takie badanie w bazie danych");
```

Comment:
Jeśli mamy cruda do Examinations i nie znajdziemy examination o podanym (jako parametr) id, powinniśmy zwrócić HttpStatusCode.NotFound ⌨️ 

To też piszę tylko tutaj, ale analogicznie powinniśmy zrobić w przypadku operacji Update, Get{id} oraz Delete